i18n-helpers/src/bin/mdbook-gettext.rs

```diff
@@ -25,27 +25,20 @@
 //! book.
 
 use anyhow::{anyhow, Context};
-use mdbook::book::Book;
 use mdbook::preprocess::{CmdPreprocessor, PreprocessorContext};
-use mdbook::BookItem;
-use mdbook_i18n_helpers::{extract_events, reconstruct_markdown, translate_events};
 use polib::catalog::Catalog;
-use polib::po_file;
 use semver::{Version, VersionReq};
-use std::{io, process};
+use std::io;
 
-fn translate(text: &str, catalog: &Catalog) -> String {
-    let events = extract_events(text, None);
-    let translated_events = translate_events(&events, catalog);
-    let (translated, _) = reconstruct_markdown(&translated_events, None);
-    translated
-}
+use mdbook_i18n_helpers::gettext::translate_book;
+use polib::po_file;
+use std::process;
 
-fn translate_book(ctx: &PreprocessorContext, mut book: Book) -> anyhow::Result<Book> {
+fn parse_catalog(ctx: &PreprocessorContext) -> anyhow::Result<Option<Catalog>> {
```

Comment:
The return type here is a bit odd to me: we can successfully parse a `None` — which seems to mean that we decided successfully decided that we don't need to parse the catalog.

Could the function be split up into a few parts? Something like this:

- `fn should_translate(...) -> bool`
- `fn load_catalog(...) -> anyhow::Result<Catalog>`
- `fn translate_book(..., catalog: &Catalog) -> Book`

Then you would only attempt to do I/O and load the catalog if it was determined that we need to do a tranlation — and you would only translate the book if you successfully loaded the catalog.

Reply:
Oh, and now that I'm in asking mode: could we get docstrings for the functions — maybe even for the ones you move around? Pretty-please? :slightly_smiling_face:

Reply:
Yeah, I agree. I refactored this a bit and decided to make `fn should_translate(ctx: &PreprocessorContext) -> anyhow::Result<bool>`. Others are OK as you suggested. Let me know what you think.